### PR TITLE
Ingress: Add detailed documentation

### PR DESCRIPTION
**Pull request by sunjayBhatia**
Details which features we support, caveats, gotchas

Fixes: https://github.com/projectcontour/contour/issues/3546
Fixes: https://github.com/projectcontour/contour/issues/3469
Updates: https://github.com/projectcontour/contour/issues/3544

### CODE REVIEW

File: site/docs/main/config/ingress.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+As TLS secrets on Ingresses are scoped to specific hosts, this default backend cannot serve TLS as it could match an unbounded set of hosts and configuring a matching set of TLS secrets would not be possible.
+As is the case on Ingress rules, Contour only supports configuring a Service as a backend and does not support any other Kubernetes resource.
+
+## Ingress Rules
+
+See upstream [documentation][4] on Ingress rules.
+
+As with default backends, Contour only supports configuring a Service as a backend and does not support any other Kubernetes resource.
+
+<!-- TODO: swap with the below once https://github.com/projectcontour/contour/pull/3381 is merged -->
```
Comment (sunjayBhatia) [inline site/docs/main/config/ingress.md:43]: Note this is commented out for now until wildcard hostname support is merged

File: site/docs/main/config/tls-delegation.md [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 In order to support wildcard certificates, TLS certificates for a `*.somedomain.com`, which are stored in a namespace controlled by the cluster administrator, Contour supports a facility known as TLS Certificate Delegation.
 This facility allows the owner of a TLS certificate to delegate, for the purposes of referencing the TLS certificate, permission to Contour to read the Secret object from another namespace.
-Delegation works for both HTTPProxy and Ingress resources.
+Delegation works for both HTTPProxy and Ingress v1beta1 resources.
```
Comment (sunjayBhatia) [inline site/docs/main/config/tls-delegation.md:5]: LMK if we should add more detail around Ingress v1 not being supported here, it is mentioned in the Ingress specific page but could be mentioned again

**Comment by sunjayBhatia** [comment]
siteproof failing due to the new page being added

File: site/docs/main/config/tls-delegation.md (hunk 2 above)
Comment (youngnick) [reply site/docs/main/config/tls-delegation.md:5]: I think some more detail would be good, yeah. Something like:
```suggestion
Delegation works for both HTTPProxy and Ingress v1beta1 resources. Ingress v1 is not currently supported due to changes in the spec that make this impossible. See <issue> for details.
```

**Review by youngnick — APPROVED** [review]
LGTM with one small suggestion.

File: site/docs/main/config/tls-delegation.md [hunk 3]
```diff
@@ -3,6 +3,8 @@
 In order to support wildcard certificates, TLS certificates for a `*.somedomain.com`, which are stored in a namespace controlled by the cluster administrator, Contour supports a facility known as TLS Certificate Delegation.
 This facility allows the owner of a TLS certificate to delegate, for the purposes of referencing the TLS certificate, permission to Contour to read the Secret object from another namespace.
 Delegation works for both HTTPProxy and Ingress v1beta1 resources.
+Ingress v1 is not currently supported due to changes in the spec that make this impossible.
```
Comment (stevesloka) [inline site/docs/main/config/tls-delegation.md:6]: nit: Maybe add something that `TLSCertificateDelegation` is currently not supported with IngressV1. My first glance it read like Ingressv1 isn't supported, but we are in the TLSCertDelegation section.

**Review by stevesloka — APPROVED** [review]
Just one small nit if you think it's needed.

File: site/docs/main/config/tls-delegation.md (hunk 3 above)
Comment (sunjayBhatia) [reply site/docs/main/config/tls-delegation.md:6]: good point, yeah ill change that

**Pull request merged**